src/model/card.ts

```diff
@@ -1,47 +1,177 @@
+
+        //Function for adding new tags to card's drop down menu
+        this.tagService.addAgent(() => {
+            this._dropDown.options.length = 0;
+            for (const t of this.tagService.tags) {
+                let option: HTMLOptionElement = document.createElement('option');
+                option.text = t.name;
+                this._dropDown.add(option);
+            }
+        });
+        this.setTag(args[0][2]);
+        
+        //Make sure the tags are updated to latest
+        this.tagService.update();
     }
 
+    /**
+     * Generates a DOM model for current card. It follows this templte:
+     * <div class="card">
+     *  <span class="sec-topic">...</>
+     *  <span class="sec-body">...</>
+     *  <button class="tag">...</>
+     *  <div class="tag-section">...</>
+     *  <button class="rembutton">...</>
+     * </>
+     */
     protected domCreator() {
         const container = document.createElement('div');
         container.className = 'cards';
 
         const displayTopic = document.createElement('span');
+        container.className = 'sec-topic';
         displayTopic.textContent = this.topic;
-        
+
         const displayBody = document.createElement('span');
+        container.className = 'sec-body';
         displayBody.textContent = this.body;
 
+        const addTagButton = document.createElement('button');
+        addTagButton.textContent = 'AddTag';
+        addTagButton.className = "addtagbutton";
+        addTagButton.onclick = () => this.addTag();
+
         const removeButton = document.createElement('button');
         removeButton.textContent = 'Delete';
         removeButton.className = "rembutton";
-
         removeButton.onclick = () => {
             container.parentNode.removeChild(container);
             this.cardService.removeCard(this);
         };
 
         container.appendChild(displayTopic);
         container.appendChild(displayBody);
+        container.appendChild(this._dropDown);
+        container.appendChild(addTagButton);
+        container.appendChild(this._tagSection);
         container.appendChild(removeButton);
 
         this._dom = container;
     }
+
+    public resetDom(){
+    }
+
+    /**  
+     * By calling this, it will associate a specific tag with
+     * this object.
+     */
+    private addTag() {
+        var x: Tag = this.tagService.valueSelect(this._dropDown.options[this._dropDown.selectedIndex].value);
+        if (this.tags.indexOf(x) < 0){
+            this.tags.push(x);
+            this.publishTag(x);
+        }
+    }
+
+    private setTag(t: string) {
+        var x: Tag = this.tagService.valueSelect(t);
+        if (this.tags.indexOf(x) < 0){
+            this.tags.push(x);
+            this.publishTag(x);
+        }
+    }
+
+    /** 
+     * This will update the tags associated with the cards and create
+     * the <span> dom under the <div class="tag-Section">.
+     * @param t 
+     */
+    private publishTag(t: Tag){
```

Comment:
instead of tracking the tag and the parent element consider using encapsulation, i.e. nested class or a separated private class.